prospero-cli/src/test/java/org/wildfly/prospero/cli/RepositoryDefinitionTest.java

```diff
@@ -75,5 +92,35 @@ public void throwsErrorIfFormatIsIncorrect() throws Exception {
         assertThrows(ArgumentParsingException.class, ()->from(List.of("foo::bar::http://test1.te")));
 
         assertThrows(ArgumentParsingException.class, ()->from(List.of("imnoturl")));
+
+    }
+
+    @Test
+    public void throwsErrorIfFormatIsIncorrectForFileURLorPathDoesNotExist() throws Exception {
+        assertThrows(ArgumentParsingException.class, ()->from(List.of("::"+fileURL)));
+
+        assertThrows(ArgumentParsingException.class, ()->from(List.of("repo-1::")));
+
+        assertThrows(ArgumentParsingException.class, ()->from(List.of("repo-1:::"+fileURL)));
+
+        assertThrows(ArgumentParsingException.class, ()->from(List.of("foo::bar::"+fileURL)));
+
+        assertThrows(ArgumentParsingException.class, ()->from(List.of("file:/path/to/repo")));
+
+        assertThrows(ArgumentParsingException.class, ()->from(List.of(filePath)));
+
+        assertThrows(ArgumentParsingException.class, ()->from(List.of("file://../repo")));
+
+    }
+
+    @Test
+    public void testCorrectRelativeOrAbsolutePathForFileURL() throws Exception {
+        try {
+            from(List.of("file:../repo"));
+            from(List.of(fileURL));
+            from(List.of("file://"+filePath));
+        } catch (Exception e) {
+            fail("Provided URL's absolute or relative path is not correct");
```

Comment:
@ChristinaDsl you don't need to catch the exception. If it fails the test fails. Also you should check that the returned value is correct. The method might run without exception, but the returned value might be wrong